docusaurus/docs/guides/securing-apis/aks-api-with-nginx-ziti-module.md

```diff
@@ -0,0 +1,369 @@
+
+- Virtual Network
+- AKS Private Cluster with Kubenet CNI and Nginx Ingress Controller
+
+Set a few environmental variables for Azure Authentication and Authorization.
+:::tip Env Vars
+export ARM_SUBSCRIPTION_ID = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+
+export ARM_CLIENT_ID = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+
+export ARM_CLIENT_SECRET = "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+
+export ARM_TENANT_ID = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+
+:::
+
+Run terraform plan.
+:::info Steps
+```shell
+git clone https://github.com/dariuszSki/openziti-nginx-ingress-terraform.git
+cd openziti-nginx-ingress-terraform/tf-provider
+terraform init
+terraform plan  -var include_aks_nginx=true  -out aks
+terraform apply "aks"
+```
+
+Once completed, grab  `cluster_public_fqdn` under `outputs` as shown in the example below.
+
+```shell
+cluster_name = "akssandeastus"
+cluster_private_fqdn = ""
+cluster_public_fqdn = "akssand-2ift1yqr.hcp.eastus.azmk8s.io"
+```
+:::
+
+## Configuration Code Snippets
+
+If you are using your own deployment method, here are some configuration details from helm chart that need to be passed to the nginx ingress controller deployment.
+
+- ***Ziti Nginx Module Identity***
+```shell
+nginx_ziti_identity = "${file("./server-nginx.json")}"
+resource "kubernetes_secret" "ziti-identity" {
+  metadata {
+    name = "nginx-ziti-identity"
+  }
+  data = {
+    "nginx-ziti-identity" = var.nginx_ziti_identity
+  }
+  type = "Opaque"
+}
+```
+
+- ***Image Reference***
+```shell
+set {
+    name = "controller.image.repository"
+    value = "docker.io/elblag91/ziti-nginx-ingress"
+  }
+
+set {
+    name = "controller.image.tag"
+    value = "3.0.2"
+}
+```
+
+- ***Configuration File - Main section block***
+:::info
+Services are commented out until they are created. Then, terraform plan can be re-run to enable them.
+:::
+```shell
+controller:
+  service:
+    create: false
+  config:
+    entries:
+      main-snippets: |
+        load_module modules/ngx_ziti_module.so;
+       
+        thread_pool ngx_ziti_tp threads=32 max_queue=65536;
+
+        #ziti identity1 {
+        #  identity_file /var/run/secrets/openziti.io/${kubernetes_secret.ziti-identity.metadata[0].name};
+
+        #  bind k8s-api {
+        #    upstream kubernetes.default:443;
+        #  }
+        #}
+```
+
+- ***Volume section and path to secrets. Added openziti.io folder.***
+```bash
+  volumes:
+      - name: "ziti-nginx-files"
+        projected:
+          defaultMode: 420
+          sources:
+          - secret:
+              name: ${kubernetes_secret.ziti-identity.metadata[0].name}
+              items: 
+              - key: ${kubernetes_secret.ziti-identity.metadata[0].name}
+                path: ${kubernetes_secret.ziti-identity.metadata[0].name}
+  volumeMounts:
+    - mountPath: /var/run/secrets/openziti.io
+      name: ziti-nginx-files
+      readOnly: true
+
+```
+
+---
+
+## OpenZiti Service Configurations

```

Comment:
It wasn't available when you drafted this, but now you could use the OpenZiti Terraform Provider, if you want.

Reply:
It's up to you whether you want to mention it, but they don't need to deploy the Ziti control plane with Terraform to use Terraform to manage their Ziti Network.